modules/scanner.py

```diff
@@ -182,81 +177,73 @@ def PortScan(target, scanspeed=5, mode=ScanMode.Normal, customflags=""):
+            log.logger(
+                "error", f"Exception: {e} raised, number of tries: {tries}"
+            )
+            continue
+        except KeyboardInterrupt:
+            raise SystemExit("Ctr+C, aborting.")
         else:
-            while True:
-                resp = nm.scan(hosts=target, arguments="-A -T 5")
-    except KeyboardInterrupt:
-        raise SystemExit("Ctr+C, aborting.")
+            break
 
 
-def NoiseScan(target, scantype=ScanType.ARP, timeout=None):
-    banner("Creating noise...", colors.green)
+def NoiseScan(target, scantype=ScanType.ARP):
+    console = Console()
+
+    banner("Creating noise...", "green")
 
+    Uphosts = TestPing(target)
     if scantype == ScanType.ARP:
         if is_root():
             Uphosts = TestArp(target)
-    else:
-        Uphosts = TestPing(target)
-
-    NoisyProcesses = []
-    for host in Uphosts:
-        info(
-            f"Started creating noise on {host}..."
-        )
-        P = Process(target=CreateNoise, args=(host,))
-        NoisyProcesses.append(P)
-        P.start()
 
     try:
-        if timeout:
-            while True:
-                print("|   " + str(timeout) + " seconds left!", end="     \r")
-                sleep(0.25)
-                print("/   " + str(timeout) + " seconds left!", end="     \r")
-                sleep(0.25)
-                print("-   " + str(timeout) + " seconds left!", end="     \r")
-                sleep(0.25)
-                print("\\   " + str(timeout) + " seconds left!", end="     \r")
-                sleep(0.25)
-                timeout -= 1
-                if timeout == 0:
-                    println("\nNoise scan complete!")
-                    for P in NoisyProcesses:
-                        P.terminate()
-                    raise SystemExit
-    except KeyboardInterrupt:
-        error("\nNoise scan interrupted!")
+        with console.status("Creating noise ...", spinner="line"):
+            NoisyProcesses = []
+            for host in Uphosts:
+                log.logger(
+                    "info", f"Started creating noise on {host}..."
+                )
+                P = Process(target=CreateNoise, args=(host,))
+                NoisyProcesses.append(P)
+                P.start()
+
+        print("Noise scan complete!")
         for P in NoisyProcesses:
             P.terminate()
+    except KeyboardInterrupt:
+        log.logger("error", "Noise scan interrupted!")
         raise SystemExit
 
 
 def DiscoverHosts(
         target,
         scantype=ScanType.ARP,
-        scanspeed=3,
         mode=ScanMode.Normal
     ):
     if isinstance(target, list):
         banner(
             f"Scanning {len(target)} target(s) using {scantype.name} scan...",
-            colors.green
+            "green"
         )
     else:
         banner(
-            f"Scanning {target} using {scantype.name} scan...", colors.green
+            f"Scanning {target} targets(s) using {scantype.name} scan...",
```

Comment:
Here we either have a single target to scan or a target subnet. So i think "target(s)" should be removed.